vllm/v1/engine/output_processor.py

```diff
@@ -180,23 +147,41 @@ def process_outputs(
     @staticmethod
     def _make_request_output(
         request_state: RequestState,
-        detokenizer_output: Optional[DetokenizerOutput],
+        new_token_ids: List[int],
+        finish_reason: Optional[str],
+        stop_reason: Optional[str],
     ) -> Optional[RequestOutput]:
 
-        if detokenizer_output is None:
+        output_kind = request_state.output_kind
+        finished = bool(finish_reason)
+        if (request_state.is_prefilling or
+            (output_kind == RequestOutputKind.FINAL_ONLY and not finished)):
```

Comment:
Probably best to always produce a request here if finished so that cleanup happens appropriately in the calling method (not sure whether this could in theory happen during prefill if the client cancels).
```suggestion
        if (request_state.is_prefilling or
            output_kind == RequestOutputKind.FINAL_ONLY) and not finished):
```